### PR TITLE
fix(testng): reworks thread locals causing TestNG Parallel issues

### LINKED ISSUE #320: TestNG driver fails when doing parallel testing
##### Issue Overview

Thread contention in the Engine class causes all sorts of different problems
This is caused by `InheritableThreadLocal` static variables in the TestNG implementation driver

##### Additional Information

PR is coming
There are two `InheritableThreadLocal` in `Arquillian.java` for TestNG:
`deployableTest` and `cycleStack`
The issues are coming up because these variables are actually shared among threads, and thus causing
issues. These need to be converted to regular `ThreadLocal`s and initialized accordingly

### PR DESCRIPTION
**Pull request by lprimak**
Fixes #320

### CODE REVIEW

**Comment by lprimak** [comment]
Payara JIRA QACI-638

**Comment by lprimak** [comment]
Any chance to get this one and #321 merged @bartoszmajsak please?

**Review by bartoszmajsak — COMMENTED** [review]
Thanks a lot, this code looks cleaner and seemingly works ;)

I just shared two comments for your consideration.

File: testng/core/src/main/java/org/jboss/arquillian/testng/Arquillian.java [hunk 1]
```diff
@@ -84,15 +89,12 @@ public void arquillianAfterSuite() throws Exception {
         }
         deployableTest.get().afterSuite();
         deployableTest.get().shutdown();
-        deployableTest.set(null);
         deployableTest.remove();
-        cycleStack.set(null);
```
[the remark this answers is not in the document]
Comment (lprimak) [reply testng/core/src/main/java/org/jboss/arquillian/testng/Arquillian.java:89]: At first I thought that was the case when I changed it, but then, when I read the docs for `ThreadLocal.remove()` it was clear that it did nothing anyway so I removed the excess code

File: testng/core/src/main/java/org/jboss/arquillian/testng/Arquillian.java [hunk 2]
```diff
@@ -206,6 +208,7 @@ public Method getMethodNewTestNG() throws ReflectiveOperationException, Security
                     return (Method) getMethod.invoke(contructorOrMethodObject);
                 }
 
+                @Override
```
[the remark this answers is not in the document]
Comment (lprimak) [reply testng/core/src/main/java/org/jboss/arquillian/testng/Arquillian.java:211]: Hmmm... usually I have the checkbox checked... I guess GH weirdness here

[the remark this answers is not in the document]
Comment (lprimak) [reply testng/core/src/main/java/org/jboss/arquillian/testng/Arquillian.java:211]: fixed

**Review by bartoszmajsak — APPROVED** [review]
Thank you! I will cut the release.

thank-you

**Comment by lprimak** [comment]
yay! thank you!

**Pull request merged**